### DISCUSSION TITLE
Pitfall of assigning named field tuple to explicit Tuple

### DISCUSSION THREAD

**Discussion by aarondel4** [dotnet/runtime#93487] [Ideas]
This was a headscratcher for me.  I made the mistake of first declaring my tuple explicitely using types, and then providing field names in the projection, not realizing the field names need to be present in the declaration as well.  But across method calls or in linq extensions I couldn't access the field names.  I started wondering if tuple field names were something that were maybe lost when passed as parameters, and that it was just some syntax sugar that only worked in local contexts.  

**Consider an Options -> C# -> Code style configuration defaulted to Suggestion that will suggest changing the declaration to include the field name or to var, _if it detects that a name field tuple is being assigned to a tuple without field names_.**

```
 IEnumerable<(Peep, Color)> peepsWithColor =
      peeps.Select(f => (
          Peep: f,
          Color: GetRandomColor()
      ));
  var bluePeeps = peepsWithColor.Where(j => j.Color == "blue").ToList(); // Error: ... does not contain a definition for "Color"
```

**Comment by huoyaoyuan** [comment]
The type of variable `peepsWithColor` is explicitly declared, without tuple element name. If you use `var` instead, the tuple element names will be preserved.

Discussion with C# tooling belongs to dotnet/roslyn repo.

**Comment by aarondel4** [comment]
> The type of variable `peepsWithColor` is explicitly declared

Yep, I stated "declaring my tuple explicitly"

> If you use `var` instead, the tuple element names will be preserved.

Yep, that's the essence of what I suggested: "suggest changing the declaration to include the field name or to var"
